optaplanner-core/src/main/java/org/optaplanner/core/impl/constructionheuristic/placer/EntityPlacerFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.constructionheuristic.placer;
+
+import org.optaplanner.core.config.constructionheuristic.placer.EntityPlacerConfig;
+import org.optaplanner.core.config.constructionheuristic.placer.PooledEntityPlacerConfig;
+import org.optaplanner.core.config.constructionheuristic.placer.QueuedEntityPlacerConfig;
+import org.optaplanner.core.config.constructionheuristic.placer.QueuedValuePlacerConfig;
+import org.optaplanner.core.impl.heuristic.HeuristicConfigPolicy;
+
+public interface EntityPlacerFactory {
+
+    static EntityPlacerFactory create(EntityPlacerConfig entityPlacerConfig) {
+        if (PooledEntityPlacerConfig.class.isAssignableFrom(entityPlacerConfig.getClass())) {
+            return new PooledEntityPlacerFactory((PooledEntityPlacerConfig) entityPlacerConfig);
+        } else if (QueuedEntityPlacerConfig.class.isAssignableFrom(entityPlacerConfig.getClass())) {
+            return new QueuedEntityPlacerFactory((QueuedEntityPlacerConfig) entityPlacerConfig);
+        } else if (QueuedValuePlacerConfig.class.isAssignableFrom(entityPlacerConfig.getClass())) {
+            return new QueuedValuePlacerFactory((QueuedValuePlacerConfig) entityPlacerConfig);
+        } else {
+            throw new IllegalArgumentException();
```

Comment:
Maybe an explanation here would be nice?

Reply:
good catch, thanks!